loader/interpolate.go

```diff
@@ -114,10 +115,14 @@ func toFloat32(value string) (interface{}, error) {
 // should match http://yaml.org/type/bool.html
 func toBoolean(value string) (interface{}, error) {
 	switch strings.ToLower(value) {
-	case "y", "yes", "true", "on":
+	case "true":
 		return true, nil
-	case "n", "no", "false", "off":
+	case "false":
 		return false, nil
+	case "y", "yes", "on":
+		return true, fmt.Errorf("%q for boolean is not supported by YAML 1.2, please use `true`", value)
```

Comment:
@ndeloof 

If the `toBoolean` function [returns err](https://github.com/compose-spec/compose-go/pull/382/files#diff-afca43eb50fd06b0f6fc2385b324e2adf3fb2b4f73915e355a9a67e000923c56R123), [`case map[string]interface{}:` of the `recursiveInterpolate` function](https://github.com/compose-spec/compose-go/blob/950902dc33e4b070107b5becff6dcb12ac790af0/interpolation/interpolation.go#L89-L91) in `interpolation/interpolation.go` will set the output to nil.

With the current structure of the code, it seems to be difficult to successfully load with warnings.

Wouldn't it be better to just fix the `case string:` in the `recursiveInterpolate` function in `interpolation/interpolation.go`, just restore the legacy boolean support, and put the warning in a separate PR?